web_responsive/static/src/js/web_responsive.js

```diff
@@ -525,4 +526,35 @@ odoo.define('web_responsive', function (require) {
     // Include the SHIFT+ALT mixin wherever
     // `KeyboardNavigationMixin` is used upstream
     AbstractWebClient.include(KeyboardNavigationShiftAltMixin);
+
+    // Document Viewer Improvements
+    DocumentViewer.include({
+        events: _.extend(DocumentViewer.prototype.events, {
+            'click .o_maximize_btn, .o_minimize_btn': '_onToggleMaximize',
+        }),
+
+        _onToggleMaximize: function (ev) {
+            ev.preventDefault();
+            var $target = $(ev.currentTarget);
+            if ($target.hasClass('o_maximize_btn')) {
+                this.$el.removeClass('o_responsive_document_viewer');
+                $target.removeClass('o_maximize_btn')
+                    .addClass('o_minimize_btn');
+                $target.find('.fa-window-maximize')
+                    .removeClass('fa-window-maximize')
+                    .addClass('fa-window-minimize')
+                    .attr('title', _("Minimize"))
+                    .attr('aria-label', _("Minimize"));
+            } else {
+                this.$el.addClass('o_responsive_document_viewer');
+                $target.addClass('o_maximize_btn')
+                    .removeClass('o_minimize_btn');
+                $target.find('.fa-window-minimize')
+                    .addClass('fa-window-maximize')
+                    .removeClass('fa-window-minimize')
+                    .attr('title', _("Maximize"))
+                    .attr('aria-label', _("Maximize"));
+            }
```

Comment:
:thinking: Have you considered adding both buttons but leaving one hidden by default, and just toggle their visibility? I think it will simplify code quite a bit... Well, actually you could even show/hide them with just CSS probably.